### PR TITLE
[WIP] Allow other modules to alter the plugin public fields info.

### PR DESCRIPTION
**Pull request by RoySegall**
#772

### CODE REVIEW

File: modules/restful_example/plugins/restful/node/articles/2.1/articles__2_1.inc [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  'name' => 'articles__2_1',
+  'entity_type' => 'node',
+  'bundle' => 'article',
+  'description' => t('Example for a plugin with altered public fields.'),
+  'class' => 'RestfulExampleArticlesResource__1_0',
+  'authentication_types' => TRUE,
+  'authentication_optional' => TRUE,
+  'major_version' => 1,
+  'minor_version' => 0,
+  'allow_public_alter' => FALSE,
```
Comment (RoySegall) [inline modules/restful_example/plugins/restful/node/articles/2.1/articles__2_1.inc:15]: => `allow_public_fields_alter`

**Comment by e0ipso** [comment]
I am not a huge fan of this. I still fail to see the need for this.

Why can't custom access callbacks in one hand, and inheritance in another help with this?

**Comment by RoySegall** [comment]
I'm trying to figure out a way for people to do as less they can by code. This mean that all the developer would need to do is to allow fields info alter and my module will handle the `access callbacks` property for them.

The module originally created to set up a backend similar to Facebook, and Twitter with the feel of third side applications, single sign on and reports for application usage: how many time the application succeeded with api access, how many times it's failed and in the far future I will add the option to notify people on this failures.

**Comment by amitaibu** [comment]
@RoySegall I think that you can solve this with a few parent classes (extending from Base, NodeBase, etc') that will have your logic.

The `drupal_alter()` overhead will indeed be too big for the value.

**Comment by RoySegall** [comment]
Closing this one. I extend and improve the API for module but I found out another problem. I'll open a follow up issue.

**Pull request closed without merging**